sim/paladin/paladin.go

```diff
@@ -123,71 +141,40 @@ func (paladin *Paladin) GetPaladin() *Paladin {
 }
 
 func (paladin *Paladin) AddRaidBuffs(raidBuffs *proto.RaidBuffs) {
-	// raidBuffs.DevotionAura = max(raidBuffs.DevotionAura, core.MakeTristateValue(
-	// 	paladin.PaladinAura == proto.PaladinAura_DevotionAura,
-	// 	paladin.Talents.ImprovedDevotionAura == 5))
-
-	// if paladin.PaladinAura == proto.PaladinAura_RetributionAura {
-	// 	raidBuffs.RetributionAura = true
-	// }
-
-	// if paladin.Talents.SanctifiedRetribution {
-	// 	raidBuffs.SanctifiedRetribution = true
-	// }
-
-	// if paladin.Talents.SwiftRetribution == 3 {
-	// 	raidBuffs.SwiftRetribution = paladin.Talents.SwiftRetribution == 3 // TODO: Fix-- though having something between 0/3 and 3/3 is unlikely
-	// }
-
-	// TODO: Figure out a way to just start with 1 DG cooldown available without making a redundant Spell
-	//if paladin.Talents.DivineGuardian == 2 {
-	//	raidBuffs.divineGuardians++
-	//}
+	if paladin.PaladinAura == proto.PaladinAura_DevotionAura {
+		raidBuffs.DevotionAura = true
+	}
+	if paladin.PaladinAura == proto.PaladinAura_RetributionAura {
+		raidBuffs.RetributionAura = true
+	}
+	if paladin.PaladinAura == proto.PaladinAura_ResistanceAura {
+		raidBuffs.ResistanceAura = true
+	}
+	if paladin.Talents.Communion {
+		raidBuffs.Communion = true
+	}
 }
 
 func (paladin *Paladin) AddPartyBuffs(_ *proto.PartyBuffs) {
 }
 
 func (paladin *Paladin) Initialize() {
-	paladin.RegisterJudgement()
-	paladin.RegisterSealOfTruth()
-	// // Update auto crit multipliers now that we have the targets.
-	// paladin.AutoAttacks.MHConfig().CritMultiplier = paladin.MeleeCritMultiplier()
-	// paladin.registerSealOfVengeanceSpellAndAura()
-	// paladin.registerSealOfRighteousnessSpellAndAura()
-	// paladin.registerSealOfCommandSpellAndAura()
-	// // paladin.setupSealOfTheCrusader()
-	// // paladin.setupSealOfWisdom()
-	// // paladin.setupSealOfLight()
-	// // paladin.setupSealOfRighteousness()
-	// // paladin.setupJudgementRefresh()
-
-	paladin.RegisterCrusaderStrike()
-	paladin.registerDivineStorm()
-
-	// paladin.registerConsecrationSpell()
-	// paladin.registerHammerOfWrathSpell()
-	// paladin.registerHolyWrathSpell()
-
-	// paladin.registerExorcismSpell()
-	// paladin.registerHolyShieldSpell()
-	// paladin.registerHammerOfTheRighteousSpell()
-	// paladin.registerHandOfReckoningSpell()
-	// paladin.registerShieldOfRighteousnessSpell()
-	// paladin.registerAvengersShieldSpell()
-	// paladin.registerJudgements()
-
-	// paladin.registerSpiritualAttunement()
-	// paladin.registerDivinePleaSpell()
-	// paladin.registerDivineProtectionSpell()
-	// paladin.registerForbearanceDebuff()
-
-	// for i := int32(0); i < paladin.Env.GetNumTargets(); i++ {
-	// 	unit := paladin.Env.GetTargetUnit(i)
-	// 	if unit.MobType == proto.MobType_MobTypeDemon || unit.MobType == proto.MobType_MobTypeUndead {
-	// 		paladin.DemonAndUndeadTargetCount += 1
-	// 	}
-	// }
+	paladin.applyGlyphs()
+	paladin.registerSpells()
+	paladin.registerPvpSets()
```

Comment:
you dont have to call an explicit register function. You can create an `init` method in your `items.go` which will automatically be called during initialization.